converting quickstart to single page article NFDEPLOY (#2636)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/quickstart/main/index.md

---
title: QuickStart — Signing in your first user
excerpt: Get started with Okta authentication
---

> In this article we will get you up and running with basic Okta authentication, quickly and effectively. This includes creating and managing Okta organizations, registering a new app and authenticating it using Okta, and next steps from there.

Reviewer: ```suggestion
In this article we get you up and running with basic Okta authentication, quickly and effectively. This includes creating and managing Okta organizations, registering a new app and authenticating it using Okta, and next steps from there.
```

---

**Learning outcomes**

* Learn what Okta organizations are, and create and manage your own organization.
* Register a sample app with your organization.

> * Authenticate it using Okta.

Reviewer: ```suggestion
* Authenticate the app using Okta.
```

**What you need**

n/a

**Sample code**

* ASP.Net — <https://github.com/okta/samples-aspnet>
* ASP.NET Core — <https://github.com/okta/samples-aspnetcore>
* ASP.Net Web Forms — <https://github.com/okta/samples-aspnet-webforms>
* Java Servlet — <https://github.com/okta/samples-java-servlet>
* Java Spring — <https://github.com/okta/samples-java-spring>
* Angular — <https://github.com/okta/samples-js-angular>
* React — <https://github.com/okta/samples-js-react>
* Vue — <https://github.com/okta/samples-js-vue>
* Golang — <https://github.com/okta/samples-golang>
* Node.js Express — <https://github.com/okta/samples-nodejs-express-4>
* Python Flask — <https://github.com/okta/samples-python-flask>
* PHP — <https://github.com/okta/samples-php>

---

## Create your Okta organization

> You built an awesome app and you want to add user authentication to it. Let’s get you set up and working with Okta.

Reviewer: ```suggestion
You built an awesome app, and you want to add user authentication to it. Let’s get you set up and working with Okta.
```

> To start, you're going to need an org. An org is a private data space &mdash; provided by Okta &mdash; that holds all the resources that you create to handle user authentication. See [Okta organizations](/docs/concepts/okta-organizations) for more information.

Reviewer: ```suggestion
To start, you're going to need an org. An org is a private data space &mdash; provided by Okta &mdash; that holds all the resources that you create to handle user authentication. See [Okta organizations](/docs/concepts/okta-organizations).
```

- **Org changes:** Displays system logs showing all the org changes in the last seven days.
- **Security Monitoring:** Displays security-related information such as recommended security tasks and suspicious user activity.

A search bar at the top of the Dashboard lets you quickly bring up information on a specific user or app.

### System Log

Some of the statistics on the Dashboard are derived from the org's system logs. You can access the System Log page either by clicking the  **View all** link in the **Org Changes** widget or the suspicious activity **View** link in the **Security Monitoring** widget. You can also access the System Log by going to **Reports** > **System Log**. The System Log records actions taken by you and any other admins, as well as ongoing events that occur for each user, such as user creation and authentication attempts. You can click each event to expand it and get full details.

In the global header in the upper-right corner of the page, you can view the Okta domain of the org that you’re in. This can be handy if you manage multiple orgs.

> The footer of the dashboard displays the version of the Okta system that you're currently running, e.g., "Version: 2020.06.4". It also displays the Okta cell your org is running in, e.g. "OK11 Cell (US)".

Reviewer: ```suggestion
The footer of the dashboard displays the version of the Okta system that you're currently running, for example: "Version: 2020.06.4". It also displays the Okta cell your org is running in, for example: "OK11 Cell (US)".
```

### Side navigation

The side navigation takes you from the Dashboard to the management pages for various aspects of your org's functionality:

| Menu          | What can you do there?                                                                             |
|---------------|----------------------------------------------------------------------------------------------------|
| Dashboard     | View your notifications, tasks, and get started guides for newbies.                                |

> | Directory     | Manage your Okta's orgs people, groups, and related attributes.                                    |

Reviewer: ```suggestion
| Directory     | Manage your Okta org's people, groups, and related attributes.                                    |
```

> | Applications  | Configure the connection between your Okta org and the apps you want to handle authentication for. |

Reviewer: ```suggestion
| Applications  | Configure the connection between your Okta org and the apps that you want to handle authentication for. |
```

| Security      | Manage the security aspects of your Okta org.                                                      |
| Workflow      | Augment Okta process flows with your own additions.                                                |
| Reports       | View your Okta org's reports and system logs.                                                      |
| Settings      | Configure org-wide settings.                                                                       |

## Add a user using the Admin Console

Any authentication system needs to have a way of keeping track of the end users who are allowed to authenticate.

> When you use Okta for authentication, you don't need to implement your own user management solution. Creating and deleting users, configuring their authentication flows, and resetting their passwords, can all be done from the Admin Console. User profiles are stored securely for you in the Okta cloud.

Reviewer: ```suggestion
When you use Okta for authentication, you don't need to implement your own user management solution. You can create and delete users, configure their authentication flows, and reset their passwords all from the Admin Console. User profiles are stored securely for you in the Okta cloud.
```

To create a new user:

1. In the Admin Console, go to **Directory** > **People**.
1. Click **Add Person**.
1. Fill in the first name, last name, username (must be an email address), and primary emails, for example:
    - **First name:** Joe
    - **Last name:** Smith
    - **Username:** joe.smith@example.com
    - **Primary email:** joe.smith@example.com

>     > **Note:** To try out end-user flows, you'll need to have access to the email address that you set for the end user.

Reviewer: ```suggestion
    > **Note:** To try out end-user flows, you need to have access to the email address that you set for the end user.
```

For Okta-hosted sign-in implementations, explore the `okta-hosted-login` folder where several use case examples are provided.

Build the example app as-is, or use it as a template for modifying your own app. When you have an app running, you're ready to connect it to your Okta org.

### Tell Okta about the app

<StackSelector snippet="register-app" />

<!--- 
#### Get values from Okta to set in the app

In the Admin Console, gather the following information from the **General** tab of your Application:

 - **Client ID**: This is the public identifier for the client, which is required for all OAuth flows.

 - **Client Secret**: This is the secret used by the client to exchange an authorization code for a token. This must be kept confidential.

Remaining in the Admin Console, go to **Security** > **API** and gather the following value found on the **Authorization Servers** tab:

 - **Issuer URI**: This is the URI of the authorization server that will perform authentication. All Developer Accounts have a "default" authorization server you can use.
 --->

### Set values in environment variables or configuration file

> After registering your app, you now have the specific values for client ID, client secret, and issuer ID required for your app integration. All the various example apps in Okta's GitHub provide ways of setting these values. Some example apps expect you to set environment variables, while others expect you to set parameters in a configuration file. Consult the README file for the example app you're using and set these three values.

Reviewer: ```suggestion
After registering your app, you now have the specific values for client ID, client secret, and issuer ID required for your app integration. All the various example apps in Okta's GitHub provide ways of setting these values. Some example apps expect you to set environment variables, while others expect you to set parameters in a configuration file. Consult the README file for the example app that you're using and set these three values.
```

## Try signing in

With the app now configured, you're ready to try signing in as an end user and see how the functionality comes together:

1. Follow the directions in the README to start the app.

2. The example apps are served by default at `localhost`. Point your browser at the URL of the app. Consult the README for default port number.

> 3. You should see a sign-in screen. Try signing in as the end user you created earlier.

Reviewer: ```suggestion
3. You should see a sign-in page. Try signing in as the end user that you created earlier.
```

For full details on API tokens, see [Create an API token](/docs/guides/create-an-api-token/create-the-token/).

### Create a user by API

To create an additional end user in your org, you can make a REST API call to the `/users` endpoint.

The full URL of the endpoint needs to begin with your Okta domain. That should be followed by the specific path for this endpoint, which is `/api/v1/users`. For example: `https://dev-133337.okta.com/api/v1/users`.

To create a new user, you make a `POST` call to that endpoint, supplying information about the new user in a JSON object in the request body.

You need to include your API token in the `Authorization` header of the request.

Here's how you would do it using curl from the command line:

```bash
curl -v -X POST \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: SSWS 00QCjAl4MlV-WPXM...0HmjFx-vbGua" \
-d '{
  "profile": {
    "firstName": "Isaac",
    "lastName": "Brock",
    "email": "isaac.brock@example.com",
    "login": "isaac.brock@example.com",
    "mobilePhone": "555-415-1337"
  }
}' "https://dev-33337.okta.com/api/v1/users?activate=false"
```

> In the above example, you'd need to replace the Okta domain in the URL with your own Okta domain. You'd also need to replace the API token in the `Authorization` header with the API token you created. Note that the value of the API token needs to be prefaced with `SSWS` followed by a space.

Reviewer: ```suggestion
In the above example, you need to replace the Okta domain in the URL with your own Okta domain. You also need to replace the API token in the `Authorization` header with the API token that you create. Note that the value of the API token needs to be prefaced with `SSWS` followed by a space.
```
